lib/Dist/Zilla/Dist/Builder.pm

```diff
@@ -462,11 +462,28 @@ might not exist.
 
 sub archive_filename {
   my ($self) = @_;
-  return join(q{},
+
+  my $name = join(q{},
     $self->dist_basename,
     ( $self->is_trial && $self->version !~ /_/ ? '-TRIAL' : '' ),
-    '.tar.gz'
   );
```

Comment:
Just an FYI while I'm reading this as a proof of concept:  when lines don't match style, I am likely to reformat or rework them to do so.  Here, for example, the use of `(...)` on join is not standard.  This is more a "hey expect this to happen" than a "I am grumpy when things are not exactly how I like them on my plate!